src/lib/pgp-key.c

```diff
@@ -457,7 +457,16 @@ pgp_seckey_t *
 pgp_decrypt_seckey(const pgp_key_t *key, FILE *passfp)
 {
     if (key->key.seckey.decrypt_cb == NULL) {
-        return NULL;
+        pgp_seckey_t *seckey = calloc(1, sizeof(*seckey));
+        if (seckey == NULL) {
+            (void) fprintf(stderr, "can't allocate memory\n");
+            return NULL;
+        }
+
+        // copy secret key for don't crash at free when the key doesn't need anymore
+        memcpy(seckey, &key->key.seckey, sizeof(*seckey));
```

Comment:
This wouldn't be sufficient unfortunately, it would just crash later in the process. (you can test this with, for example, `rnp --sign somefile` with an unencrypted G10 key).
The best way to handle this is at the call-site. If the key is not encrypted, this function shouldn't even be called.
I've been working on something like that as part of #39 and #396 but it's not quite ready yet.